tracer/src/Datadog.FeatureFlags.OpenFeature/DatadogProvider.cs

```diff
@@ -0,0 +1,154 @@
+// <copyright file="DatadogProvider.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace Datadog.FeatureFlags.OpenFeature;
+
+/// <summary>
+/// OpenFeature V2.0.0+ Provider for Datadog
+/// </summary>
+public class DatadogProvider : global::OpenFeature.FeatureProvider
+{
+    private Metadata _metadata = new Metadata("datadog-openfeature-provider");
+
+    /// <summary> Gets provider metadata </summary>
+    /// <returns> Returns provider metadata </returns>
+    public override Metadata? GetMetadata() => _metadata;
+
+    /// <summary> Resolve flag as boolean </summary>
+    /// <param name="flagKey"> Requested flag </param>
+    /// <param name="defaultValue"> Default value </param>
+    /// <param name="context"> Evaluation context </param>
```

Comment:
I was going to complain about the docs, and then realised the OpenFeature docs are just as bad 😂